code-challenges.js

```diff
@@ -9,67 +9,207 @@
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
 
-
+const vowelRem2 = (input) => {
+  if(typeof input === "string"){
+    let stringLetters = input.split("")
+    return stringLetters.filter(letter => {
+      return letter !== "a" && letter !== "A" && letter !== "e" && letter !== "E" && letter !== "i" && letter !== "I" && letter !== "o" && letter !== "O" && letter !== "u" && letter !== "U"
+    }).join("")}
+  else{
+    return `${input} is not a string`
+  }
+}
+console.log(vowelRem2(refactorTester1));
+console.log(vowelRem2(refactorTester2));
+console.log(vowelRem2(refactorTester3));
 
 
 
 // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
 // Expected output: "chopsticks" "stemware"
-
-var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
-var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
-
-
-
+//
+//Pseudocode
+//create a function
+//argument will be an array
+//use .map and .length to create new array with lengths of strings inside as numbers
+//extract first element from new array
+//compare length to previous record length
+//if larger record index and length of long word
+//if smaller ignore
+//repeat cycle until all elements have been compared
+//return the recorded biggest index
+//console.log(function(array)); to test
+//
+ var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
+ var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
+
+const bigString = (array) => {
+  var bigNumIndex = 0
+  var bigNum = 0
+  var arrayIndex = array.map( value => {
+    return value.length
+  })
+  for(i = 0; i < array.length; i++){
+    if(arrayIndex[i] > bigNum){
+      bigNum = arrayIndex[i]
+      bigNumIndex = i;
+    }
+  }
+  return array[bigNumIndex]
+}
+
+console.log(bigString(utensils));
+console.log(bigString(vessels));
 
 
 // --------------------7) Create a function that takes in an array of strings and returns an array with the last letter removed from each string.
 // Expected output: ["Luig", "Peac", "Toa", "Yosh"] ["Ink", "Blink", "Pink", "Clyd"]
 
+//Pseudocode
+//Create a function
+//argument will be an array of strings
+//start loop
+//extract first element
+//break element (using .split) into an array of characters
+//remove last character element from array using .pop
+//recombine character array into a single string using .join
+//add string to new array using .push
+//loop recycles
+//end loop after array.length cycles
+//return new array with shortened strings
+//console.log(function(array)); to test
+
 var mario = ["Luigi", "Peach", "Toad", "Yoshi"]
 var pacman = ["Inky", "Blinky", "Pinky", "Clyde"]
+
+const lostLetter = (array) => {
+  let splitChar = []
+  let connectedChar = []
+  for(i = 0; i < array.length; i++){
+    splitChar = array[i].split("")
+    splitChar.pop()
+    connectedChar.push(splitChar.join(""))
+  } return connectedChar
```

Comment:
```suggestion
  } 
  return connectedChar
```
Generally you'd put this return statement on the next line